### PR TITLE
lightning/importinto: fix insert err after import for AUTO_ID_CACHE=1 and SHARD_ROW_ID_BITS

### LINKED ISSUE #52654: Compatible issue when using lightning + SHARD_ROW_ID_BITS + AUTO_ID_CACHE=1
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. 
```
CREATE TABLE t (
  id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (id) /*T![clustered_index] NONCLUSTERED */
) /*T![auto_id_cache] AUTO_ID_CACHE=1 */ /*T! SHARD_ROW_ID_BITS=4 PRE_SPLIT_REGIONS=3 */;

INSERT INTO t VALUES (1778961125641936898);
```

2. dumpling the data then drop table t and using lightning physical mode to import data.

3. `insert into t values ()`

### 2. What did you expect to see? (Required)
No error, because we can continues insert rows like,
```
CREATE TABLE t (
  id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (id) /*T![clustered_index] NONCLUSTERED */
) /*T![auto_id_cache] AUTO_ID_CACHE=1 */ /*T! SHARD_ROW_ID_BITS=4 PRE_SPLIT_REGIONS=3 */;

INSERT INTO t VALUES (1778961125641936898);
INSERT INTO t VALUES ();
```

### 3. What did you see instead (Required)
```
mysql> insert into t values ();
ERROR 1467 (HY000): Failed to read auto-increment value from storage engine
```

### 4. What is your TiDB version? (Required)
5bb8ed75efa5f1f6b52277c64916663adc6947a5

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #52654

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

```
mysql> CREATE TABLE t (
    ->   id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
    ->   PRIMARY KEY (id) /*T![clustered_index] NONCLUSTERED */
    -> ) /*T![auto_id_cache] AUTO_ID_CACHE=1 */ /*T! SHARD_ROW_ID_BITS=4 PRE_SPLIT_REGIONS=3 */;
Query OK, 0 rows affected (0.09 sec)

mysql>
mysql> import into t from select 1778961125641936898;
Query OK, 1 row affected (0.30 sec)
Records: 1, ID: 0e8a4adb-47df-4790-a237-5a489e25d9c6

mysql> insert into t values ();
Query OK, 1 row affected (0.01 sec)

mysql> select *, _tidb_rowid from t;
+---------------------+---------------------+
| id                  | _tidb_rowid         |
+---------------------+---------------------+
| 1778961125641936898 | 1152921504606846977 |
| 1778961125641936899 | 2882303761517117442 |
+---------------------+---------------------+
2 rows in set (0.01 sec)
mysql> select *, _tidb_rowid & x'07FFFFFFFFFFFFFF' from t;
+---------------------+-----------------------------------+
| id                  | _tidb_rowid & x'07FFFFFFFFFFFFFF' |
+---------------------+-----------------------------------+
| 1778961125641936899 |                                 2 |
| 1778961125641936898 |                                 1 |
+---------------------+-----------------------------------+
2 rows in set (0.01 sec)
```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: lightning/pkg/importer/table_import.go [hunk 1]
```diff
@@ -953,26 +953,30 @@ func (tr *TableImporter) postProcess(
...
-			if err == nil && isLocalBackend(rc.cfg) {
+			if isLocalBackend(rc.cfg) {
 				// for TiDB version >= 6.5.0, a table might have separate allocators for auto_increment column and _tidb_rowid,
 				// especially when a table has auto_increment non-clustered PK, it will use both allocators.
 				// And in this case, ALTER TABLE xxx AUTO_INCREMENT = xxx only works on the allocator of auto_increment column,
 				// not for allocator of _tidb_rowid.
 				// So we need to rebase IDs for those 2 allocators explicitly.
-				err = common.RebaseGlobalAutoID(ctx, adjustIDBase(newBase), tr, tr.dbInfo.ID, tr.tableInfo.Core)
+				err = common.RebaseTableAllocators(ctx, map[autoid.AllocatorType]int64{
+					autoid.RowIDAllocType:    tr.alloc.Get(autoid.RowIDAllocType).Base(),
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply lightning/pkg/importer/table_import.go:970]: no need add 1, it's last used value

**Review by lance6716 — APPROVED** [review]

**Pull request merged**